src/Charts/ChataColumnChart.js

```diff
@@ -0,0 +1,508 @@
+import { select } from 'd3-selection'
+import { ChataChartListPopover } from './ChataChartListPopover'
+import { ChataChartSeriesPopover } from './ChataChartSeriesPopover'
+
+import {
+    enumerateCols,
+    getIndexesByType,
+    getMetadataElement,
+    makeGroups,
+    getMinAndMaxValues,
+    formatLabel,
+    getVisibleSeries,
+    toggleSerie
+} from './ChataChartHelpers'
+import {
+    SCALE_BAND,
+    SCALE_LINEAR,
+    getAxisBottom,
+    getAxisLeft,
+    setDomainRange,
+    getBandWidth,
+    getColorScale,
+    getLegend
+} from './d3-compatibility'
+import {
+    formatColumnName,
+    getStringWidth,
+    formatData,
+    formatChartData,
+    closeAllChartPopovers,
+} from '../Utils'
+import { tooltipCharts } from '../Tooltips'
+
+export function createColumnChart(
+    component, json, options, onUpdate=()=>{}, fromChataUtils=true,
+    valueClass='data-chartindex', renderTooltips=true){
+    var margin = {
+        top: 15,
+        right: 10,
+        bottom: 60,
+        left: 90,
+        marginLabel: 50,
+        bottomChart: 50
+    },
+    width = component.parentElement.clientWidth - margin.left;
+    var height;
+
+    var cols = enumerateCols(json);
+    var indexList = getIndexesByType(cols);
+    var xIndexes = [];
+    var yIndexes = [];
+    let chartWidth;
+    var legendOrientation = 'horizontal';
+    var shapePadding = 100;
+    const legendBoxMargin = 15;
+
+    if(indexList['STRING']){
+        xIndexes.push(...indexList['STRING'])
+    }
+
+    if(indexList['DATE']){
+        xIndexes.push(...indexList['DATE'])
+    }
+
+    if(indexList['DATE_STRING']){
+        xIndexes.push(...indexList['DATE_STRING'])
+    }
+
+    if(indexList['DOLLAR_AMT']){
+        yIndexes = indexList['DOLLAR_AMT'];
+    }else if(indexList['QUANTITY']){
+        yIndexes = indexList['QUANTITY'];
+    }
+
+    var metadataComponent = getMetadataElement(component, fromChataUtils);
+    if(!metadataComponent.metadata){
+        metadataComponent.metadata = {
+            groupBy: {
+                index: xIndexes[0].index,
+                currentLi: 0,
+            },
+            series: yIndexes
+        }
+    }
+
+    var xAxisIndex = metadataComponent.metadata.groupBy.index;
+    var activeSeries = metadataComponent.metadata.series;
+    var data = makeGroups(json, options, activeSeries, cols[xAxisIndex].index);
+    const minMaxValues = getMinAndMaxValues(data);
+    var index1 = activeSeries[0].index;
+    var index2 = cols[xAxisIndex].index;
+
+    var colStr1 = cols[index2]['display_name'] || cols[index2]['name'];
+    var colStr2 = cols[index1]['display_name'] || cols[index1]['name'];
+
+    var col1 = formatColumnName(colStr1);
+    var col2 = formatColumnName(colStr2);
+    const barWidth = width / data.length;
+    const rotateLabels = barWidth < 135;
+    const interval = Math.ceil((data.length * 16) / width);
+    var xTickValues = [];
+    var allLengths = [];
+    if (barWidth < 16) {
```

Comment:
I noticed that some of the code formatting is not consistent. Do you use any linting or auto-formatters?  I highly recommend using prettier or eslint. Personally I have mine format the file every time I save so I never have to think about it.